### DISCUSSION TITLE
Error Reading URLs

### DISCUSSION THREAD

**Discussion by rmonty7321** [martinpaljak/nfc4pc#11] [General]
First: Great tool! I appreciate all of the documentation and help you provided!

This might be user error, but whenever I try to read NTAG215 tags using an HID OmniKey 5022 USB reader, running version 23.07.10 of the program on both MacOS and Windows 10, I obtain the error "# WARNING: No URL found in tag with UID #, use -u to specify UID URL" (which is expected if the tags did not have URLs written on them; "UID #" is replaced with the unique UID tag identifier for the specific tag whose URL I am attempting to read and open). 

However, I have written, rewritten, and re-rewritten URLs on multiple tags, using different NFC writing tools, only to see this same error message when attempting to open them with the NFC4PC program. I also double-checked that the URLs are correctly written by successfully reading and opening them on an iPhone (however, I do recognize iPhone NFC software integration is quite different than this program on Windows/MacOS). All other NFC4PC program functionality, including browser navigation and specific argument lines work perfectly!

Do you have any specific instructions on 1) URL formats for this program (i.e. is there a different format to follow than what is common on NFC writing programs like NFC Tools or NFC21 Tools) or 2) any workaround to read/open these written URLs? Thanks!

**Comment by martinpaljak** [comment]
Can you provide a debug log or maybe just a screenshot from a phone of the tag as read by NFC tag info or similar tool? If you add a title to the URL you write, it gets converted into a smart poster, and those are explicitly not supported (but you should get a warning about that specific use)

**Comment by rmonty7321** [comment]
Absolutely! Here are screenshots from the program while debugging and from
a phone with the NFC tag info.

**Reply by martinpaljak** [reply]
You missed pics

**Comment by rmonty7321** [comment]
![Attachment1](https://github.com/martinpaljak/nfc4pc/assets/111897563/3dae5d87-52e0-435a-aa9c-10dca026fb89)
![Attachment0](https://github.com/martinpaljak/nfc4pc/assets/111897563/fa3a14ef-5311-46cf-9063-a3e2785fb510)

Apologies!

**Reply by martinpaljak** [reply]
This looks like a bug/parsing issue, will look at why the header parsing fails.

**Reply by martinpaljak** [reply]
Could you give this build a spin: https://github.com/martinpaljak/nfc4pc/actions/runs/8478097132/artifacts/1369085659

**Reply by martinpaljak** [reply]
as I think this was a bug that already caught my attention during testing but has not turned into a release. https://github.com/martinpaljak/nfc4pc/commit/1523e2f38e3f23e545fbe4568ec5a38442fcb481

**Reply by rmonty7321** [reply]
Works perfectly! Thank you so much for the time and effort!

**Reply by martinpaljak** [reply]
OK. I'll prepare a new release at some point, but there is no ETA.
